frontend/src/components/common/AlertNotification.tsx

```diff
@@ -0,0 +1,113 @@
+import { Box, useTheme } from '@material-ui/core';
+import React from 'react';
+import { useRouteMatch } from 'react-router-dom';
+import { testAuth } from '../../lib/k8s/apiProxy';
+import { getRoutePath, ROUTES } from '../../lib/router';
+import { useTypedSelector } from '../../redux/reducers/reducers';
+
+const NOT_FOUND_ERROR_MESSAGE = 'Error: Api request error: Bad Gateway';
+
+// in ms
+const NETWORK_STATUS_CHECK_TIME = 5000;
+
+export default function AlertNotification(){
+  const routes = useTypedSelector(state => state.ui.routes);
+  const [networkStatusCheckTimeFactor, setNetworkStatusCheckTimeFactor] = React.useState(0);
+  const [error, setError] = React.useState<null | string | boolean>(null);
+  const [intervalID, setIntervalID] = React.useState<NodeJS.Timeout | null>(null);
+
+  function registerSetInterval(): NodeJS.Timeout {
+    return setInterval(() => {
+      setError(null);
+      testAuth().then(() => {
+        setError(false);
+      })
+        .catch((err) => {
+          const error = new Error(err);
+          setError(error.message);
+          setNetworkStatusCheckTimeFactor(networkStatusCheckTimeFactor =>
+            networkStatusCheckTimeFactor + 1);
```

Comment:
Why is this using a function here? Shouldn't we just use `++networkStatusCheckTimeFactor`?

Reply:
as setState is asynchronous we can't rely on networkStatusCheckTimeFactor to be the new state value whenever we are incrementing, The callback to setState ensures that it is always called with the latest state value